From c903f66194f04e97fc684f5a9654cedb27530931 Mon Sep 17 00:00:00 2001
From: Ludwig Krispenz <lkrispen@redhat.com>
Date: Mon, 31 Jul 2017 10:51:08 +0200
Subject: [PATCH 1/3] Ticket 49334 - fix backup restore if changelog exists

The corrcect flag to copy a directory in backup/restore must be passed for the changelog directory

Reviewed by: William, thanks
---
 ldap/servers/slapd/back-ldbm/dblayer.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ldap/servers/slapd/back-ldbm/dblayer.c b/ldap/servers/slapd/back-ldbm/dblayer.c
index ff97aa4..3a97f2f 100644
--- a/ldap/servers/slapd/back-ldbm/dblayer.c
+++ b/ldap/servers/slapd/back-ldbm/dblayer.c
@@ -6143,7 +6143,7 @@ dblayer_backup(struct ldbminfo *li, char *dest_dir, Slapi_Task *task)
             return_value = dblayer_copy_directory(li, task, changelogdir,
                                                   changelog_destdir,
                                                   0 /* backup */,
-                                                  &cnt, 1, 0, 0);
+                                                  &cnt, 0, 0, 1);
             if (return_value) {
                 slapi_log_err(SLAPI_LOG_ERR,
                           "dblayer_backup", "Error in copying directory "
@@ -6823,7 +6823,7 @@ int dblayer_restore(struct ldbminfo *li, char *src_dir, Slapi_Task *task, char *
                     *cldirname = '\0';
                     return_value = dblayer_copy_directory(li, task, filename1,
                                                           changelogdir, 1 /* restore */,
-                                                          &cnt, 1, 0 ,0);
+                                                          &cnt, 0, 0 ,1);
                     *cldirname = '/';
                     if (return_value) {
                         slapi_log_err(SLAPI_LOG_ERR,
-- 
2.9.4

